File: BSharp/ClientApp/src/app/data/dto/get-by-id-response.ts
import { EntityWithKey } from '../entities/base/entity-with-key';
import { EntityForSave } from '../entities/base/entity-for-save';
import { Entity } from '../entities/base/entity';

export class GetByIdResponse<TEntity extends EntityWithKey = EntityWithKey> {
  Result: TEntity;
  CollectionName: string;
  RelatedEntities: { [key: string]: EntityWithKey[]; };
}

File: Tellma.Api.Web.Dto/GetByIdResponse.cs
using System;
using System.Collections.Generic;
using Tellma.Model.Common;

namespace Tellma.Api.Dto
{
    public class GetByIdResponse<TEntity> where TEntity : Entity
    {
        public Dictionary<string, object> Extras { get; set; }

        public TEntity Result { get; set; }

        public string CollectionName { get; set; }
        
        public Dictionary<string, IEnumerable<EntityWithKey>> RelatedEntities { get; set; }

        public DateTimeOffset ServerTime { get; set; }
    }
}
